config/src/lib.rs

```diff
@@ -1257,9 +1257,16 @@ pub struct Config {
 
     #[serde(default = "default_true")]
     pub allow_download_protocols: bool,
+
+    #[serde(default = "default_local_echo_threshold_ms")]
+    pub local_echo_threshold_ms: u64,
```

Comment:
What do you think about scoping this per domain?
The code change is a little more involved:

Move this config value so that it lives over here alongside this one:
https://github.com/wez/wezterm/blob/4efcc81bee162579888d418f152b55550b7b8503/config/src/tls.rs#L81

Then add something like this:

```rust
   pub local_echo_threshold_ms: Option<u64>,
```

to `ClientInner`:
https://github.com/wez/wezterm/blob/4efcc81bee162579888d418f152b55550b7b8503/wezterm-client/src/domain.rs#L21

That will need to get initialized, so add that same optional value as a parameter to:
https://github.com/wez/wezterm/blob/4efcc81bee162579888d418f152b55550b7b8503/wezterm-client/src/domain.rs#L155

to do that, I think you'll want to add an accessor method for this value over here, similar to this name method:
https://github.com/wez/wezterm/blob/4efcc81bee162579888d418f152b55550b7b8503/wezterm-client/src/domain.rs#L122-L123

so that you can pass it down through here:
https://github.com/wez/wezterm/blob/4efcc81bee162579888d418f152b55550b7b8503/wezterm-client/src/domain.rs#L510

then with that in place, the `should_predict` method could be phrased as something like this:

```rust
fn should_predict(&self) -> bool {
  self.client.local_echo_threshold_ms.map(|thresh| self.last_input_rtt >= thresh).unwrap_or(false)
}
```

Reply:
Aha, yes, that sounds like a good idea. And that would perhaps even allow us to drop the `is_local()` machinery in favour of a different default for unix domains?
I'll have a go at it, probably next week though.